torchx/schedulers/ray/ray_driver.py

```diff
@@ -7,6 +7,7 @@
 import json
```

Comment:
would be good to add a longish docstring explaining the elastic scheduling algorithm and all of the edge cases and how they're handled

Reply:
done